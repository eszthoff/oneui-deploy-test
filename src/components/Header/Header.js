import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import styles from './Header.scss';

const { block, elem } = bem({
    name: 'Header',
    classnames: styles
});

const Header = props => {
    const { logoSrc, logoLink, logoTitle, children, ...rest } = props;

    return (
        <div {...rest} {...block(props)}>
            <div {...elem('wrapper', props)}>
                <a href={logoLink}>
                    <img src={logoSrc} alt={logoTitle} {...elem('logo', props)} />
                </a>
                <div {...elem('menu', props)}>{children}</div>
            </div>
        </div>
    );
};

Header.displayName = 'Header';

Header.propTypes = {
    /** path to the logo source file */
    logoSrc: PropTypes.string.isRequired,
    /** path to where the logo should link to if clicked */
    logoLink: PropTypes.string,
    /** a name for the logo, it will used as alternative text to the img */
    logoTitle: PropTypes.string,
    /** Node(s) to be rendered on the right side of the header */
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node])
};

Header.defaultProps = {
    logoLink: '/',
    logoTitle: 'website logo',
    children: null
};

export default Header;
