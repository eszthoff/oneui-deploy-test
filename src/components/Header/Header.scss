@import '../../themes/oneui/oneui.scss';

.Header {
    width: 100%;
    height: 50px;
    border-bottom: 1px solid var(--border-color-strong);
    box-shadow: 0 1px 3px var(--border-color-light);

    &__wrapper {
        box-sizing: border-box;
        max-width: var(--site-container-size);
        height: 100%;
        padding: var(--spacing-normal);
        margin: auto;
        display: flex;
        justify-content: space-between;
        align-items: center;
        align-content: center;
    }

    &__logo {
        height: 40px;
    }
}
