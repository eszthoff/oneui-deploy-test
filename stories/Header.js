import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { text, withKnobs } from '@storybook/addon-knobs'; // eslint-disable-line import/no-extraneous-dependencies
import { Header } from '@textkernel/oneui';

storiesOf('Header', module)
    .addDecorator(withKnobs)
    .add('Header', () => (
        <Header
            logoSrc={text('Image URL', 'https://www.jobfeed.nl/images/jobfeed-logo.svg')}
            logoLink={text('URL where click on logo will lead', '/')}
            logoName={text('Alt text for the logo', 'Jobfeed')}
        >
            {text('Left side', 'This is a placeholder for children')}
        </Header>
    ));
