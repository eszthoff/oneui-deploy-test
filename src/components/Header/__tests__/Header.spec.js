import React from 'react';
import toJson from 'enzyme-to-json';
import Header from '../Header';

describe('Header component that renders a website header with a logo on the left and optional children on the right side', () => {
    it('should render correctly', () => {
        const wrapper = mount(
            <Header logoSrc="jobfeed-logo.svg">This is a placeholder for children</Header>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
