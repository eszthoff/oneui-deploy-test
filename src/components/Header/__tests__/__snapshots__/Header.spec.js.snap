// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component that renders a website header with a logo on the left and optional children on the right side should render correctly 1`] = `
<Header
  logoLink="/"
  logoSrc="jobfeed-logo.svg"
  logoTitle="website logo"
>
  <div
    className="Header"
  >
    <div
      className="Header__wrapper"
    >
      <a
        href="/"
      >
        <img
          alt="website logo"
          className="Header__logo"
          src="jobfeed-logo.svg"
        />
      </a>
      <div
        className="Header__menu"
      >
        This is a placeholder for children
      </div>
    </div>
  </div>
</Header>
`;
